### DISCUSSION TITLE
使用Mysql JDBC 驱动如何获取doris内部真实数据类型？

### DISCUSSION THREAD

**Discussion by mengzhisy** [apache/doris#17732] [A - General / Q&A]
```sql
select cast(1 as LARGEINT);
```
对于这个sql，Doris内部返回类型是largeint，但是JDBC驱动返回的是schema信息是string类型。我应该如何获取真实的数据类型，我需要知道这个类型是int还是string。

**Comment by ya-dao** [comment]
JDBC驱动应该暂时是没法获取本就没有定义的类型的, 比如doris里面的datetime v2这种. 下面两种方法也许能帮上忙: 
* 可以考虑在doris提供的那些api里面找找看有没有能用的. 
   * Table Schema Action(看注释可能不是用于这种用途的, 需要确认一下)
   * Query Schema Action(和下面的sql查询是一样的效果)
* sql的话可以考虑下通过desc ${tableName}的结果进行处理
